packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -32,6 +32,8 @@ export class EngineReportingExtension<TContext = any>
   implements GraphQLExtension<TContext> {
   private treeBuilder: EngineReportingTreeBuilder;
   private explicitOperationName?: string | null;
+  private gqlParsingSucceeded?: boolean;
```

Comment:
I'm kind of assuming a new instance of this extension is created on every request due to the `queryString` and `documentAST` fields here, but I was a little lost in the constructor logic, truth be told.

Reply:
Correct, that is the lifecycle.